crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -32,23 +32,59 @@ pub enum SuiteKind {
     Class,
 
     /// Statements in any other body (e.g., `if` or `while`).
-    #[default]
-    Other,
+    Other {
+        /// Whether this suite is the last suite in the current statement.
+        ///
+        /// Below, `last_suite_in_statement` is `false` for the suite containing `foo10` and `foo12`
+        /// and `true` for the suite containing `bar`.
+        /// ```python
+        // if sys.version_info >= (3, 10):
+        //     def foo10():
+        //         return "new"
+        // elif sys.version_info >= (3, 12):
+        //     def foo12():
+        //         return "new"
+        // else:
+        //     def bar():
+        //         return "old"
+        // ```
+        last_suite_in_statement: bool,
+    },
 }
 
-#[derive(Debug)]
-pub struct FormatSuite {
-    kind: SuiteKind,
+impl Default for SuiteKind {
+    fn default() -> Self {
+        Self::Other {
+            // For stability, we can't insert an empty line if we don't know if the outer suite
+            // also does.
+            last_suite_in_statement: true,
+        }
+    }
 }
 
```

Comment:
I find this a surprising default. I think we should remove it and update the remaining call sites to be explicit about it.

Reply:
We unfortunately need a default for `AsFormat`, otherwise i'm all for making this explicit in call sites.

Reply:
Hmm, that's unfortunate but fair enough.